src/ui/server/controllers/insurance/account/create/verify-email/index.ts

```diff
@@ -0,0 +1,39 @@
+import api from '../../../../../api';
+import { ROUTES } from '../../../../../constants';
+import { Request, Response } from '../../../../../../types';
+
+const {
+  INSURANCE: {
+    ACCOUNT: { CREATE, SIGN_IN },
+  },
+  PROBLEM_WITH_SERVICE,
+} = ROUTES;
+
+/**
+ * get
+ * Call the API to verify the token passed in the request query
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} Sign in or link expired page
+ */
+export const get = async (req: Request, res: Response) => {
+  try {
+    const { token } = req.query;
+
+    if (token) {
+      const response = await api.keystone.account.verifyEmailAddress(token);
+
+      if (response.success) {
+        return res.redirect(SIGN_IN.ROOT);
+      }
+
+      return res.redirect(CREATE.VERIFY_EMAIL_LINK_EXPIRED);
+    }
+
+    return res.redirect(CREATE.VERIFY_EMAIL_LINK_EXPIRED);
```

Comment:
if there's no token, would it still be expired or problem with service?

Reply:
Good spot, I had a good old think about this. If no token is provided, ideally I think from a user/design perspective we would redirect to a "no token provided" or "invalid/broken URL, try ...." kind of page.

Redirecting to "problem with service" would be misleading because there isn't actually a problem, just the token/URL sent by the user is invalid (or doesn't exist).

I think this is an ok solution for MVP unless anyone else thinks otherwise. i'll raise with design for post MVP improvement.

Reply:
Discussed with design, this is all fine for MVP, design will look into better messaging/pages for these scenarios for post MVP 💪